app/lib/features/chat/widgets/custom_input.dart

```diff
@@ -325,9 +326,65 @@ class _CustomChatInputState extends ConsumerState<CustomChatInput> {
                             );
                         inputNotifier.setEditMessage(message);
                         if (message is TextMessage) {
-                          ref
-                              .read(_textValuesProvider(roomId).notifier)
-                              .update((state) => message.text);
+                          // Parse String Data to HTML document
```

Comment:
uhh... not a big fan of parsing html ourselves to be honest ... but okay if this must be done ... but can we move that into a separate function and add unit-tests for it to make sure it always works the way we want?!?

Reply:
not a big fan of the "add unit-tests later"-approach that usually doesn't happen ... it really isn't much of a deal to be honest: once you have a simple input-output fn [adding some tests](https://docs.flutter.dev/cookbook/testing/unit/introduction) throwing various inputs against it and checking the output shouldn't be much of a hassle...

Reply:
@gnunicorn
Added uni test cased for edit message with user mentions feature. :)

Also uploaded video in this PR description which shows test case execution.